frontend/components/CompactNavigationBar.js

```diff
@@ -0,0 +1,440 @@
+import React, { useEffect, useState } from "react";
+import { useRouter } from "next/router";
+import Link from "next/link";
+import Image from "next/image";
+import styles from "../styles/Navbar.module.css";
+
+// import menu icons
+import MenuIcon from "@material-ui/icons/Menu";
+import CloseIcon from "@material-ui/icons/Close";
+// import menuDropdownContainer icons
+import ChromeReaderModeOutlinedIcon from "@material-ui/icons/ChromeReaderModeOutlined";
+import FavoriteBorderIcon from "@material-ui/icons/FavoriteBorder";
+import PeopleOutlineIcon from "@material-ui/icons/PeopleOutline";
+import PhoneOutlinedIcon from "@material-ui/icons/PhoneOutlined";
+import InsertDriveFileOutlinedIcon from "@material-ui/icons/InsertDriveFileOutlined";
+import HeadsetMicOutlinedIcon from "@material-ui/icons/HeadsetMicOutlined";
+import LocalPrintshopOutlinedIcon from "@material-ui/icons/LocalPrintshopOutlined";
+import ImageOutlinedIcon from "@material-ui/icons/ImageOutlined";
+import VideocamOutlinedIcon from "@material-ui/icons/VideocamOutlined";
+
+const CompactNavigationBar = () => {
+  const [clearNavBar, setClearNavBar] = useState(true);
+  const [hiddenNavBar, setHiddenNavBar] = useState(false);
+  const [menuOpen, setMenuOpen] = useState(false);
+  const [aboutUsDropdownOpen, setAboutUsDropdownOpen] = useState(false);
+  const [mediaDropdownOpen, setMediaDropdownOpen] = useState(false);
+  const [darkMode, setDarkMode] = useState(false);
+  const router = useRouter();
+  const pageLoading = false;
+
+  /* check whether device is in light or dark mode 
```

Comment:
NIT: comments throughout this PR should be one liners i.e. using `//` syntax rather than `/* */` . If we require using the latter syntax, think about whether the comment itself could be simplified.

Also consider whether it is necessary to leave comments. Sometimes if we have good variable names, it should be clear as to what the code is trying to achieve.